pytoda/datasets/_csv_lazy_dataset.py

```diff
@@ -29,37 +42,47 @@ def __init__(
         Args:
             filepath (str): path to .csv file.
             feature_list (FeatureList): a list of features. Defaults to None.
-            chunk_size (int): size of the chunks. Defauls to 10000.
+            chunk_size (int): size of the chunks. Defaults to 10000.
             kwargs (dict): additional parameters for pd.read_csv.
                 Except from nrows and chunksize.
         """
         self.chunk_size = chunk_size
-        _CacheDataset.__init__(self)
-        _CsvDataset.__init__(
+
+        size_limit = 1073741824  # default limit of 1GiB from diskcash
+        file_size = os.path.getsize(filepath)
+        if file_size > size_limit:
```

Comment:
swapping this check with the constructor of the `_CacheDatasource` might make sense. Then we wouldn't need to hardcode the size limit.

Reply:
I added  to the `_CacheDatasource` constructor the option to pass a filepath to fit.